src/main/java/net/imagej/ops/OpSignature.java

```diff
@@ -0,0 +1,188 @@
+
+package net.imagej.ops;
+
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.scijava.module.DefaultMutableModuleItem;
+import org.scijava.module.ModuleInfo;
+import org.scijava.module.ModuleItem;
+
+import net.imglib2.IterableInterval;
+import net.imglib2.RandomAccessible;
+import net.imglib2.RealRandomAccessible;
+import net.imglib2.type.numeric.RealType;
+
+/**
+ * The minimum required components for a call to
+ * {@link OpService#run(String, Object...)}
+ *
+ * @author Gabriel Selzer
+ */
+public class OpSignature {
+
+	private final String name;
+	private final String functionalType;
+	private final List<ModuleItem<?>> inputs;
+	private final List<ModuleItem<?>> outputs;
+
+	public OpSignature(final OpInfo info) {
+		this(info.getName(), info.getType(), info.inputs(), info.outputs());
+	}
+
+	public OpSignature(final String name, final Class<?> functionalType,
+		final List<ModuleItem<?>> inputs, final List<ModuleItem<?>> outputs)
```

Comment:
I dislike the `List<ModuleItem<?>>` inputs; it doesn't seem like the `OpSignature` should know about them (or about `Module`s in general). But they are a container for so much data we need, and it's so convenient to store the data this way (so that we can call `inputsFor`, for example.

Reply:
We no longer use the `List<ModuleItem<?>>`s for input and output.